mne/decoding/base.py

```diff
@@ -746,3 +758,86 @@ def _make_scorer(scoring):
+    inverse_func = [False]
+    if hasattr(estimator, 'steps'):
+        # if pipeline, retrieve all steps by nesting
+        inverse_func = list()
+        for _, est in estimator.steps:
+            inverse_func.extend(_get_inverse_funcs(est, terminal=False))
+    elif hasattr(estimator, 'inverse_transform'):
+        # if not pipeline attempt to retrieve inverse function
+        inverse_func = [estimator.inverse_transform]
+
+    # If terminal node, check that that the last estimator is a classifier,
+    # and remove it from the transformers.
+    if terminal:
+        last_is_estimator = inverse_func[-1] is False
+        all_invertible = not(False in inverse_func[:-1])
+        if last_is_estimator and all_invertible:
+            # keep all inverse transformation and remove last estimation
+            inverse_func = inverse_func[:-1]
+        else:
+            inverse_func = list()
+
+    return inverse_func
+
+
+def get_coef(estimator, attr='filters_', inverse_transform=False):
+    """Retrieve the coefficients of an estimator ending with a Linear Model.
+
+    This is typically useful to retrieve "spatial filters" or "spatial
+    patterns" of decoding models [1]_.
+
+    Parameters
+    ----------
+    estimator : object | None
+        An estimator from scikit-learn.
+    attr : str
+        The name of the coefficient attribute to retrieve. Typically 'filters_'
+        or 'patterns_'. Defaults to 'filters_'.
+    inverse_transform : bool
+        If True, returns the coefficients after inverse transforming them with
+        the transformer steps of the estimator.
+
+    Returns
+    -------
+    coef : array
+        The coefficients.
+
+    References
+    ----------
+    .. [1] Haufe, S., Meinecke, F., Gorgen, K., Dahne, S., Haynes, J.-D.,
+       Blankertz, B., & Biessmann, F. (2014). On the interpretation of weight
+       vectors of linear models in multivariate neuroimaging. NeuroImage, 87,
+       96-110. doi:10.1016/j.neuroimage.2013.10.067.
+    """
+    # If searchlight, loop across estimators
+    if hasattr(estimator, 'estimators_'):
+        coef = list()
+        for est in estimator.estimators_:
+            coef.append(get_coef(est, attr, inverse_transform))
+        return np.array(coef)
+
+    else:
+        # Get the coefficients of the last estimator in case of nested pipeline
+        est = estimator
+        while hasattr(est, 'steps'):
+            est = est.steps[-1][1]
+        if not hasattr(est, attr):
+            raise ValueError('This estimator does not have a %s '
+                             'attribute.' % attr)
+        coef = getattr(est, attr)
+
+        # inverse pattern e.g. to get back physical units
+        if inverse_transform:
+            if not hasattr(estimator, 'steps'):
+                raise ValueError('inverse_transform can only be applied onto '
+                                 'pipeline estimators.')
+
+            for inverse_func in _get_inverse_funcs(estimator)[::-1]:
```

Comment:
I'd add a comment to make explicit we want to loop in reverse in order to reverse the direction of model fitting